docs/basic-features/environment-variables.md

```diff
@@ -47,7 +47,7 @@ export async function getStaticProps() {
 ```
 
 > **Note**: In order to keep server-only secrets safe, Next.js replaces `process.env.*` with the correct values
-> at build time. This means that `process.env` is not a standard JavaScript object, so you’re not able to
+> at run time. This means that `process.env` is not a standard JavaScript object, so you’re not able to
 > use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).
 > Environment variables must be referenced as e.g. `process.env.PUBLISHABLE_KEY`, _not_ `const { PUBLISHABLE_KEY } = process.env`.
```

Comment:
This line seems inaccurate we only replace `NEXT_PUBLIC_` prefixed values otherwise the values will only be available if set in the current env.

It might be better to replace this section with:
> **Note**: In order to keep server-only secrets safe, Next.js does not inline them in the bundles like `NEXT_PUBLIC_` values and they must be available in the environment `next` is started in to be accessible.